### PR TITLE
[DIA-2024] Ship ProGuard rules with the release AAR

### PR DESCRIPTION
**Pull request by bohdan-go-wombat**
JIRA - [DIA-2024](https://sourcepoint.atlassian.net/browse/DIA-2024)

For release AAR there are shipped proguard rules:
image

For other env AARs (for example, debug one) there are no shipped rules:
image

[DIA-2024]: https://sourcepoint.atlassian.net/browse/DIA-2024?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by carmelo-iriti** [comment]
The pr looks good and we need to edit the README file in the proguard section where the rules are listed. The clients have to know that it is not necessary anymore to add the rules in their proguard file  because now they are included into the aar

**Comment by carmelo-iriti** [comment]
To test this feature you should:
- generate a release aar artifact
- create a new android project 
- add the aar in the lib folder
- and execute a property 
If you can see the message the aar works

N.B. The app in the new project has to be signed with a new Keystore and in the release variant has to include the `minifyEnabled true` gradle option

File: .github/workflows/proguard-test.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: SDK Release
```
Comment (carmelo-iriti) [inline .github/workflows/proguard-test.yml:1]: before merging we can remove this file

File: cmplibrary/gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-VERSION_NAME = 7.6.1
+VERSION_NAME = dia-2024
```
Comment (carmelo-iriti) [inline cmplibrary/gradle.properties:1]: before merging we can revert this change

File: .github/workflows/proguard-test.yml (hunk 1 above)
Comment (bohdan-go-wombat) [reply .github/workflows/proguard-test.yml]: yup, done

File: cmplibrary/gradle.properties (hunk 2 above)
Comment (bohdan-go-wombat) [reply cmplibrary/gradle.properties]: yup, done

File: README.md [hunk 3]
```diff
@@ -644,7 +644,10 @@ After adding the `Privacy Manager Id for the Property Group`, you should set the
 
 ## ProGuard
 
-Using ProGuard in your project you might need to add the following rules
+**From version 7.6.2 CMP library ships it's own Proguard rules with the AAR**, so the user of the library don't have to add anything manually. Make sure you are up-to-date with the latest releases of the CMP library.
```
Comment (Nevazhnovu) [inline README.md:647]: Thanks for the readme update

File: cmplibrary/cmp-consumer-proguard-rules.pro [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Proguard rules for CMP consumers
```
Comment (Nevazhnovu) [inline cmplibrary/cmp-consumer-proguard-rules.pro:1]: @bohdan-go-wombat by any chance, is it possible to change the cmp's log level in consumer release using proguard rules file?

Comment (carmelo-iriti) [reply cmplibrary/cmp-consumer-proguard-rules.pro:1]: @Nevazhnovu the proguard rules aren’t related with the SDK log messages

**Review by carmelo-iriti — APPROVED** [review]

File: cmplibrary/cmp-consumer-proguard-rules.pro (hunk 4 above)
Comment (bohdan-go-wombat) [reply cmplibrary/cmp-consumer-proguard-rules.pro:1]: @Nevazhnovu yes, @carmelo-iriti is right, the logs are not related to the proguard rules, but can you elaborate more on what you wanted? maybe we can figure it out, but we need more context on what we want to achieve
